src/main/java/io/github/moulberry/notenoughupdates/profileviewer/level/LevelPage.java

```diff
@@ -0,0 +1,197 @@
+import java.util.Collections;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class LevelPage {
+
+	private final GuiProfileViewer instance;
+	private final BasicPage basicPage;
+	private ProfileViewer.Profile profile;
+
+	private final int randomNumber;
+
+	private final int randomNumber2;
+
+	private String profileId;
+
+	private int guiLeft, guiTop;
+
+	private JsonObject constant;
+
+	private final CoreTaskLevel coreTaskLevel;
+	private final DungeonTaskLevel dungeonTaskLevel;
+	private final EssenceTaskLevel essenceTaskLevel;
+	private final MiscTaskLevel miscTaskLevel;
+	private final SkillRelatedTaskLevel skillRelatedTaskLevel;
+	private final SlayingTaskLevel slayingTaskLevel;
+	private final StoryTaskLevel storyTaskLevel;
+
+	private static final ResourceLocation pv_levels = new ResourceLocation("notenoughupdates:pv_levels.png");
+
+	public LevelPage(GuiProfileViewer instance, BasicPage basicPage) {
+		this.instance = instance;
+		this.basicPage = basicPage;
+		constant = Constants.SBLEVELS;
+
+		coreTaskLevel = new CoreTaskLevel(this);
+		dungeonTaskLevel = new DungeonTaskLevel(this);
+		essenceTaskLevel = new EssenceTaskLevel(this);
+		miscTaskLevel = new MiscTaskLevel(this);
+		skillRelatedTaskLevel = new SkillRelatedTaskLevel(this);
+		slayingTaskLevel = new SlayingTaskLevel(this);
+		storyTaskLevel = new StoryTaskLevel(this);
+
+
+		ThreadLocalRandom current = ThreadLocalRandom.current();
+		randomNumber = current.nextInt(0, 69 + 69 + 69 + 69 + 69 + 69 + 7);
+		randomNumber2 = current.nextInt(0, 2);
+	}
+
+	public void drawPage(int mouseX, int mouseY, float partialTicks) {
+		this.guiLeft = GuiProfileViewer.getGuiLeft();
+		this.guiTop = GuiProfileViewer.getGuiTop();
+		this.profile = GuiProfileViewer.getProfile();
+		this.profileId = GuiProfileViewer.getProfileId();
+
+		basicPage.drawSideButtons();
+
+		if (constant == null) {
+			Utils.showOutdatedRepoNotification();
+			return;
+		}
+
+		Minecraft.getMinecraft().getTextureManager().bindTexture(pv_levels);
+		Utils.drawTexturedRect(guiLeft, guiTop, instance.sizeX, instance.sizeY, GL11.GL_NEAREST);
+
+		double skyblockLevel = profile.getSkyblockLevel(profileId);
+		JsonObject profileInfo = profile.getProfileInformation(profileId);
+
+		if (randomNumber == (69 + 69 + 69 + 69 + 69 + 69 + 6)) {
+			iVMMXVII(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		}
+		drawMainBar(skyblockLevel, mouseX, mouseY, guiLeft, guiTop);
+		coreTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		dungeonTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		essenceTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		miscTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		skillRelatedTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		slayingTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+		storyTaskLevel.drawTask(profileInfo, mouseX, mouseY, guiLeft, guiTop);
+	}
+
+	private void iVMMXVII(
```

Comment:
What? please use readable names for methods and parameters

Reply:
No its a secret for a reason

Reply:
its supposed to look obfuscated. i dont like it either tho

Reply:
I agree. this is the wrong way to do a secret.